addons/partner_autocomplete/models/res_partner.py

```diff
@@ -130,6 +132,25 @@ def read_by_vat(self, vat, timeout=15):
         if vies_vat_data:
             return [self._format_data_company(vies_vat_data)]
         else:
+            try:
+                vies_result = check_vies(vat)
+                if vies_result and vies_result['valid'] and (name := vies_result['name']) != '---':
+                    address = list(filter(bool, vies_result['address'].split('\n')))
+                    street = address[0]
+                    zip_city = address[-1].split(' ', 1) if len(address) > 1 else [None, None]
+                    street2 = address[1] if len(address) > 2 else None
+                    return [self._iap_replace_location_codes({
+                        'name': name,
+                        'vat': vat,
+                        'street': street,
+                        'street2': street2,
+                        'city': zip_city[1],
+                        'zip': zip_city[0],
+                        'country_code': vies_result['countryCode'],
+                        'skip_enrich': True,
+                    })]
+            except Exception:
+                _logger.exception("Failed VIES VAT check.")
```

Comment:
Only the call to `check_vies` should be in the scope of the `try ... except ...`.
As we're using a broad `except Exception`, it could hide legitimate errors.